hypothesis-python/src/hypothesis/strategies/_internal/numbers.py

```diff
@@ -182,97 +179,94 @@ def integers(
 )
 
 
+def _sign_aware_lte(x: float, y: float) -> bool:
+    """Less-than-or-equals, but strictly orders -0.0 and 0.0"""
+    if x == 0.0 == y:
+        return math.copysign(1.0, x) <= math.copysign(1.0, y)
+    else:
+        return x <= y
+
+
 class FloatStrategy(SearchStrategy):
-    """Generic superclass for strategies which produce floats."""
+    """A strategy for floating point numbers."""
 
-    def __init__(self, allow_infinity, allow_nan, allow_subnormal, width):
+    def __init__(
+        self,
+        min_value: Real = -math.inf,
+        max_value: Real = math.inf,
```

Comment:
These should definitely be `float` annotations - we convert other types in `st.floats()`, and it's useful to have the exact bounds here.

I'd expect to also need an `allow_subnormal` flag sometimes?  But maybe not yet...

Reply:
ACK on the float annotations.

`allow_subnormal` could be part of the mid-level interterface I think. Another decision I thought about but don't understand the consequences of: is it better to filter the subnormals out, or to instead cast them into normals, like I'm doing with widths. (assuming it's not part of this interface)